chain/network-primitives/src/types.rs

```diff
@@ -347,26 +345,36 @@ impl KnownPeerStatus {
 pub struct KnownPeerState {
     pub peer_info: PeerInfo,
     pub status: KnownPeerStatus,
-    pub first_seen: u64,
-    pub last_seen: u64,
+    first_seen: Time,
+    last_seen: Time,
```

Comment:
Sorry for back and forth here, but I feel this should be 

```rust
    pub first_seen: Time,
    pub last_seen: Time,
```

and the getters/setters should be removed.

That is 

If we store raw `u64` internally, we should make fields private and provide well-typed accessor functions, to not expose raw representation. 

If we store `Time` internally, we should just make everything pub, because `fn foo(&self) / fn set_foo(&mut self)` pair is strictly worse than just making the field pub.